### ISSUE TITLE
linking error when compiling for iOS with xcodebuild 14.3

### ISSUE THREAD

**Issue by codenashwan** [flutter/flutter#124433]
### Is there an existing issue for this?

- [ ] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [ ] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

```
$ flutter run
```
```
Launching lib/main.dart on iPhone 14 Pro Max in debug mode...
main.dart:1
Upgrading Pods-Runner-frameworks.sh
Xcode build done.                                           16.1s
Failed to build iOS app
Error (Xcode): File not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Could not build the application for the simulator.
Error launching application on iPhone 14 Pro Max.
Exited
```

### Actual results

```
Launching lib/main.dart on iPhone 14 Pro Max in debug mode...
main.dart:1
Upgrading Pods-Runner-frameworks.sh
Xcode build done.                                           16.1s
Failed to build iOS app
Error (Xcode): File not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Could not build the application for the simulator.
Error launching application on iPhone 14 Pro Max.
Exited
```

### Flutter Doctor output

```
Last login: Sat Apr  8 03:49:47 on ttys001
codenashwan@Nashwans-MBP ~ % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.10, on macOS 13.2.1 22D68 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.77.1)
[✓] Connected device (1 available)
[✓] HTTP Host Availability

• No issues found!
codenashwan@Nashwans-MBP ~ % 


```

**Comment by CoMatu** [comment]
In APPNAME/ios/App/Pods/Target Support Files/Pods-App/Pods-App-frameworks.sh replace source="$(readlink "${source}")" with source="$(readlink -f "${source}")"

**Comment by m3talsmith** [comment]
@CoMatu this doesn't work for me.

Step 3 breaks because there is no config variable in the post_install section of the Podfile.

Step 4 falls apart because:

- The path is different: `APPNAME/ios/Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh`
- The file already has that change in there

This continues to be an issue that points towards sqlite header files missing. The issue is unresolved in my case.

**Comment by vasil-boshnyakov** [comment]
```
In APPNAME/ios/App/Pods/Target Support Files/Pods-App/Pods-App-frameworks.sh replace source="$(readlink "${source}")" with source="$(readlink -f "${source}")"
```

the folder `Pods` is specified in `.gitignore`. We need a proper solution please.

**Comment by ahmed-alhelali** [comment]
is there any solution yet? i'm facing the same issue (:

**Comment by alexdess** [comment]
Same issue

**Comment by millerf** [comment]
Same issue

**Comment by exaby73** [comment]
Hello. Is this issue reproducible with the default counter app? Which plugins are being used here? Could you provide the contents on your `pubspec.yaml` file?

**Comment by millerf** [comment]
I made it work!

Following https://stackoverflow.com/questions/75894992/missing-file-libarclite-iphoneos-a-in-xcode-14-3-after-update-flutter-and-xcod

Updated  `PodFile` to have this at the end:
```
post_install do |installer|
  installer.generated_projects.each do |project|
    project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'
         end
    end
end
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```

Now the build works.


Note: It is probably the same solution that @m3talsmith ([here](https://github.com/flutter/flutter/issues/124433#issuecomment-1501172617) ), but I didn't know exactly where to add that line.

**Comment by ahmed-alhelali** [comment]
> I made it work!
> 
> Following https://stackoverflow.com/questions/75894992/missing-file-libarclite-iphoneos-a-in-xcode-14-3-after-update-flutter-and-xcod
> 
> Updated `PodFile` to have this at the end:
> 
> ```
> post_install do |installer|
>   installer.generated_projects.each do |project|
>     project.targets.each do |target|
>         target.build_configurations.each do |config|
>             config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'
>          end
>     end
> end
>   installer.pods_project.targets.each do |target|
>     flutter_additional_ios_build_settings(target)
>   end
> end
> ```
> 
> Now the build works.
> 
> Note: It is probably the same solution that @m3talsmith ([here](https://github.com/flutter/flutter/issues/124433#issuecomment-1501172617) ), but I didn't know exactly where to add that line.

thanks a lot! it did work for me and the build works (:

**Comment by codenashwan** [comment]
> Hello. Is this issue reproducible with the default counter app? Which plugins are being used here? Could you provide the contents on your `pubspec.yaml` file?


I checked my packages thoroughly, I saw some packages have problems,
Those that ✅ work without problems on the new version and those that ❌ do not work and shows the error that I posted



```
name: shop
description: A new Flutter project.

publish_to: 'none'

version: 1.0.0+1

environment:
  sdk: '>=2.18.6 <3.0.0'

dependencies:
❌  cached_network_image: ^3.2.3 
✅  cupertino_icons: ^1.0.5
✅ flutter:
✅   sdk: flutter
✅  geolocator: ^9.0.2
✅  get: ^4.6.5
✅   get_storage: ^2.1.1
❌ google_maps_flutter: ^2.2.5
❌ google_sign_in: ^6.1.0
✅ intl: ^0.17.0


dev_dependencies:
  flutter_lints: ^2.0.0
  flutter_test:
    sdk: flutter

flutter:

  uses-material-design: true

  assets:
    - assets/img/
    - assets/img/intro/


```

**Comment by exaby73** [comment]
@codenashwan Were you able to fix it with the solution provided in https://github.com/flutter/flutter/issues/124433#issuecomment-1501525956?

**Comment by exaby73** [comment]
I meant adding `config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'` to your `Podfile` which should not be in `.gitignore`

**Comment by codenashwan** [comment]
> I meant adding `config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'` to your `Podfile` which should not be in `.gitignore`

No, I haven't tried it

**Comment by jmagman** [comment]
@exaby73 please let's not suggest workarounds like adding `IPHONEOS_DEPLOYMENT_TARGET` to Podfiles in this issue tracker, we want to fix the issue for everyone in Flutter itself without any workarounds being necessary, and it makes it harder to track down the right fix. 🙂 

In this case, the originally reported `libarclite_iphonesimulator` issue was tracked in https://github.com/flutter/flutter/issues/124340 and has been hotfixed to Flutter 3.7.11.  Please remove any workaround you added to your Podfile and run `flutter upgrade` to get this fix.

**Comment by jmagman** [comment]
And there were some additional comments in this issue about updating `readlink` in a CocoaPods script, and that's completely unrelated to the originally reported `libarclite_iphonesimulator` issue.  That is tracked in  https://github.com/flutter/flutter/issues/123890 and has been fixed in Flutter 3.7.10.

**Comment by exaby73** [comment]
@jmagman Thanks for the insight :) I'll keep this in mind going forward
